modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/PostalAddressResourceImpl.java

```diff
@@ -106,6 +130,225 @@ public Page<PostalAddress> getUserAccountPostalAddressesPage(
+	}
+
+	@Override
+	public PostalAddress putAccountPostalAddress(
+			Long accountId, Long postalAddressId, PostalAddress postalAddress)
+		throws Exception {
+
+		Address address = _validateAccountPostalAddres(
+			accountId, postalAddressId);
+
+		Country country = _getCountryByTitle(postalAddress);
+
+		long regionId = _getRegionByTitleAndCountry(postalAddress, country);
+
+		ListType type = _getType(postalAddress);
+
+		address = _addressLocalService.updateAddress(
+			address.getAddressId(), postalAddress.getName(),
+			address.getDescription(), postalAddress.getStreetAddressLine1(),
+			postalAddress.getStreetAddressLine2(),
+			postalAddress.getStreetAddressLine3(),
+			postalAddress.getAddressLocality(), postalAddress.getPostalCode(),
+			regionId, country.getCountryId(), type.getListTypeId(),
+			address.isMailing(), postalAddress.getPrimary(),
+			address.getPhoneNumber());
+
+		return PostalAddressUtil.toPostalAddress(
+			contextAcceptLanguage.isAcceptAllLanguages(), address,
+			contextCompany.getCompanyId(),
+			contextAcceptLanguage.getPreferredLocale());
+	}
+
+	private Country _getCountryByTitle(PostalAddress postalAddress) {
+		List<Country> countries = _countryService.getCompanyCountries(
+			contextCompany.getCompanyId());
+
+		Iterator<Country> countryIterator = countries.iterator();
+
+		Boolean found = false;
+
+		String title = null;
+
+		Country country = null;
+
+		while (countryIterator.hasNext() && !found) {
+			country = countryIterator.next();
+
+			title = country.getTitle(
+				contextAcceptLanguage.getPreferredLocale());
+
+			if (title.equals(postalAddress.getAddressCountry())) {
+				found = true;
+			}
+		}
+
+		if (!found) {
+			throw new BadRequestException("Country not found");
+		}
+
+		return country;
+	}
+
+	private long _getRegionByTitleAndCountry(
+		PostalAddress postalAddress, Country country) {
+
+		if (postalAddress.getAddressType() == null) {
+			return 0;
+		}
+
+		List<Region> regions = _regionService.getRegions(
+			country.getCountryId());
+
+		if (regions.isEmpty()) {
+			return 0;
+		}
+
+		Iterator<Region> regionIterator = regions.iterator();
+
+		Region region = null;
+		String title = null;
+
+		Boolean found = false;
+
+		while (regionIterator.hasNext() && !found) {
+			region = regionIterator.next();
+
+			title = region.getTitle(
+				contextAcceptLanguage.getPreferredLanguageId());
+
+			if (title.equals(postalAddress.getAddressRegion())) {
+				found = true;
+			}
+		}
+
+		if (!found) {
+			throw new BadRequestException("Region not found");
+		}
+
+		return region.getRegionId();
+	}
+
+	private ListType _getType(PostalAddress postalAddress) {
+		ListType type = _listTypeLocalService.getListType(
+			postalAddress.getAddressType(),
+			"com.liferay.account.model.AccountEntry.address");
+
+		if (type == null) {
+			throw new BadRequestException("Type not found");
+		}
+
+		return type;
+	}
+
+	private Address _validateAccountPostalAddres(
```

Comment:
Typo: "Address" not "Addres"

Reply:
Changed! ([commit](https://github.com/liferay-appsec/liferay-portal/pull/1415/commits/325ca15db6050a31f3971929fb4618a4b6a3f609))